tests/ImageSharp.Benchmarks/PixelBlenders/PorterDuffBulkVsSingleVector.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using BenchmarkDotNet.Attributes;
+using SixLabors.ImageSharp.PixelFormats.PixelBlenders;
+
+namespace SixLabors.ImageSharp.Benchmarks.PixelBlenders;
+
+public class PorterDuffBulkVsSingleVector
+{
+    private Vector4[] backdrop;
+    private Vector4[] source;
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        this.backdrop = new Vector4[8 * 20];
+        this.source = new Vector4[8 * 20];
+
+        FillRandom(this.backdrop);
+        FillRandom(this.source);
+    }
+
+    private static void FillRandom(Vector4[] arr)
+    {
+        Random rng = new();
+        for (int i = 0; i < arr.Length; i++)
+        {
+            arr[i].X = rng.NextSingle();
+            arr[i].Y = rng.NextSingle();
+            arr[i].Z = rng.NextSingle();
+            arr[i].W = rng.NextSingle();
+        }
+    }
+
+    [Benchmark(Description = "Scalar", Baseline = true)]
+    public Vector4 OverlayValueFunction_Scalar()
+    {
+        Vector4 result = default;
+        for (int i = 0; i < this.backdrop.Length; i++)
+        {
+            result = PorterDuffFunctions.NormalSrcOver(this.backdrop[i], this.source[i], .5F);
```

Comment:
The only concern with these benchmarks is that - unlike [the real span code](https://github.com/SixLabors/ImageSharp/blob/c06da8c4bcf37cbdc327a90d38bedbbff08fca73/src/ImageSharp/PixelFormats/PixelBlenders/DefaultPixelBlenders.Generated.tt#L129) - they do not walk the `source` and `amount` buffers, meaning there are fewer cache misses. 

I would have created a parametric benchmark (N=2,16,128,512,1024) that uses the Span API-s, and run it against the code before/after the changes. Shouldn't significantly impact the result, but would paint a more nuanced picture about the improvements.

Anyways, amazing job!

Reply:
They actually used to walk those buffers. I was surprised by that, so I replicated it in the benchmark.

https://github.com/SixLabors/ImageSharp/blob/f53c78d7570c06eca02030e24d9c104e20a5431c/src/ImageSharp/PixelFormats/PixelBlenders/DefaultPixelBlenders.Generated.tt#L87-L103

I have actually realized that the benchmarks were actually slightly generous to the old approach as I'd forgotten the clamp on `amount`